src/backend/src/services/work-packages.services.ts

```diff
@@ -701,4 +702,103 @@ export default class WorkPackagesService {
 
     return workPackageTemplateTransformer(workPackage);
   }
+
+  static async editWorkPackageTemplate(
+    submitter: User,
+    workPackageTemplateId: string,
+    templateName: string,
+    templateNotes: string,
+    duration: number | undefined,
+    stage: WorkPackageStage | undefined,
+    blockedBy: BlcockedByCreateArgs[],
+    expectedActivities: string[],
+    deliverables: string[],
+    workPackageName: string | undefined
+  ): Promise<Work_Package_Template> {
+    // retrieving original work package template
+    const originalWorkPackageTemplate = await prisma.work_Package_Template.findUnique({
+      where: { workPackageTemplateId },
+      include: { blockedBy: true }
+    });
+
+    // error handling
+    if (!originalWorkPackageTemplate) throw new NotFoundException('Work Package Template', workPackageTemplateId);
+    if (originalWorkPackageTemplate.dateDeleted) throw new DeletedException('Work Package Template', workPackageTemplateId);
+    if (!isAdmin(submitter.role)) throw new AccessDeniedAdminOnlyException('edit work package templates');
+
+    // retrieving list of blocked By ids in work package template
+    const blockedByIds = originalWorkPackageTemplate.blockedBy.map((item) => item.blockedByInfoId);
+
+    // retrieving list of blocked By ids in work package template
+    const newBlockedByIds = blockedBy.map((item) => item.blockedByInfoId);
+
+    // checking differences in both blocked by lists
+    for (const blockedByItemId of blockedByIds) {
+      const blockedBy = await prisma.blocked_By_Info.findUnique({
+        where: {
+          blockedByInfoId: blockedByItemId
+        }
+      });
+
+      // deleting a blocked by if the new list does not contain it
+      if (!newBlockedByIds.includes(blockedByItemId)) {
+        await prisma.blocked_By_Info.delete({
+          where: {
+            blockedByInfoId: blockedByItemId
+          }
+        });
+      }
+
+      // updating otherwise
+      else {
+        await prisma.blocked_By_Info.update({
+          where: {
+            blockedByInfoId: blockedByItemId
+          },
+          data: {
+            name: blockedBy?.name,
+            stage: blockedBy?.stage
+          }
+        });
+      }
+    }
+
+    // checking for new blocked by in the new list
+    const isNewBlockedBy = (blockedByItem: BlcockedByCreateArgs) => {
+      return !originalWorkPackageTemplate.blockedBy.some(
+        (oldItem) => oldItem.blockedByInfoId === blockedByItem.blockedByInfoId
+      );
+    };
+
+    // filtering blocked by list to those that are new
+    const blockedByToCreate = blockedBy.filter(isNewBlockedBy);
```

Comment:
oh ok I see, so my other comment will basically be moving this up where you're making the other lists (I'd just do it in one line, I don't think the helper method is necessary in this case), and then having a different line for creating, deleting, and updating, for readability sake